From 3ff7f2b035c1f7eedd8d861de8ef1417a8a6f58e Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Tue, 14 Mar 2017 09:39:53 +0800
Subject: [PATCH] arm64: dts: rockchip: add cpu-avs device node for rk3368

Add cpu-avs node in the device tree for the ARM64 rk3368 SoC.

Change-Id: Ie7eee09c20b06bd755b9277e0acd8eaf810c5331
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3368.dtsi | 25 ++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368.dtsi b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
index 905818e2a5ff..ac1c4cdff882 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
@@ -258,6 +258,31 @@
 		};
 	};
 
+	cpu_avs: cpu-avs {
+		cluster0-avs {
+			cluster-id = <0>;
+			min-volt = <950000>; /* uV */
+			min-freq = <216000>; /* KHz */
+			leakage-adjust-volt = <
+			/*  mA        mA         uV */
+			    0         254        0
+			>;
+			nvmem-cells = <&cpu_leakage>;
+			nvmem-cell-names = "cpu_leakage";
+		};
+		cluster1-avs {
+			cluster-id = <1>;
+			min-volt = <950000>; /* uV */
+			min-freq = <216000>; /* KHz */
+			leakage-adjust-volt = <
+			/*  mA        mA         uV */
+			    0         254        0
+			>;
+			nvmem-cells = <&cpu_leakage>;
+			nvmem-cell-names = "cpu_leakage";
+		};
+	};
+
 	arm-pmu {
 		compatible = "arm,armv8-pmuv3";
 		interrupts = <GIC_SPI 112 IRQ_TYPE_LEVEL_HIGH>,
-- 
2.35.3

